sky/provision/gcp/instance.py

```diff
@@ -145,11 +151,17 @@ def terminate_instances(
     handler_to_instances = _filter_instances(handlers, project_id, zone,
                                              label_filters, lambda _: None)
     operations = collections.defaultdict(list)
-    for handler, instances in handler_to_instances.items():
-        for instance in instances:
-            operations[handler].append(
-                handler.terminate(project_id, zone, instance))
-    _wait_for_operations(operations, project_id, zone)
+    try:
+        for handler, instances in handler_to_instances.items():
+            for instance in instances:
+                operations[handler].append(
+                    handler.terminate(project_id, zone, instance))
+        _wait_for_operations(operations, project_id, zone)
+    except errors.HttpError as e:
+        if not allow_resource_not_found:
+            raise
+        if _RESOURCE_NOT_FOUND_PATTERN.search(e.reason) is None:
+            raise
```

Comment:
In what case will the exception be caught?

It seems in L151, we already query the instances. If the instance does not exist, the `handler_to_insstances` will be empty?

Reply:
Also, should we catch the exception per operation inside the `_wait_for_operations`, instead of globally? Otherwise, the later operations will not be waited if a former operation fails.